src/main/scala/org/apache/spark/deploy/armada/submit/ArmadaClientApplication.scala

```diff
@@ -211,42 +187,311 @@ private[spark] class ArmadaClientApplication extends SparkApplication {
       conf.get(ARMADA_SPARK_EXECUTOR_LABELS).map(commaSeparatedLabelsToMap).getOrElse(Map.empty)
 
     val armadaClientUrl = conf.get("spark.master")
-    val armadaClusterUrl =
-      if (conf.get(ARMADA_SERVER_INTERNAL_URL).nonEmpty)
-        s"local://armada://${conf.get(ARMADA_SERVER_INTERNAL_URL)}"
-      else {
-        armadaClientUrl
+    val armadaClusterUrl = conf
+      .get(ARMADA_SERVER_INTERNAL_URL)
+      .filter(_.nonEmpty)
+      .map { internalUrl =>
+        s"local://armada://$internalUrl"
       }
+      .getOrElse(armadaClientUrl)
 
-    ArmadaJobConfig(
+    // Parse resource limits and requests without defaults
+    val driverLimitCores      = conf.get(ARMADA_DRIVER_LIMIT_CORES)
+    val driverRequestCores    = conf.get(ARMADA_DRIVER_REQUEST_CORES)
+    val driverLimitMemory     = conf.get(ARMADA_DRIVER_LIMIT_MEMORY)
+    val driverRequestMemory   = conf.get(ARMADA_DRIVER_REQUEST_MEMORY)
+    val executorLimitCores    = conf.get(ARMADA_EXECUTOR_LIMIT_CORES)
+    val executorRequestCores  = conf.get(ARMADA_EXECUTOR_REQUEST_CORES)
+    val executorLimitMemory   = conf.get(ARMADA_EXECUTOR_LIMIT_MEMORY)
+    val executorRequestMemory = conf.get(ARMADA_EXECUTOR_REQUEST_MEMORY)
+
+    CLIConfig(
       queue = queue,
       jobSetId = jobSetId,
       namespace = conf.get(ARMADA_SPARK_JOB_NAMESPACE),
-      priority = conf.get(ARMADA_SPARK_JOB_PRIORITY),
+      priority = Some(conf.get(ARMADA_SPARK_JOB_PRIORITY)),
       containerImage = containerImage,
       podLabels = podLabels,
       driverLabels = driverLabels,
       executorLabels = executorLabels,
       nodeSelectors = nodeSelectors.getOrElse(Map.empty),
       nodeUniformityLabel = gangUniformityLabel,
-      armadaClusterUrl = armadaClusterUrl,
-      executorConnectionTimeout = Duration(conf.get(ARMADA_EXECUTOR_CONNECTION_TIMEOUT), SECONDS)
+      armadaClusterUrl = Some(armadaClusterUrl),
+      executorConnectionTimeout =
+        Some(Duration(conf.get(ARMADA_EXECUTOR_CONNECTION_TIMEOUT), SECONDS)),
+      driverLimitCores = driverLimitCores,
+      driverRequestCores = driverRequestCores,
+      driverLimitMemory = driverLimitMemory,
+      driverRequestMemory = driverRequestMemory,
+      executorLimitCores = executorLimitCores,
+      executorRequestCores = executorRequestCores,
+      executorLimitMemory = executorLimitMemory,
+      executorRequestMemory = executorRequestMemory
     )
   }
 
-  private[spark] case class ArmadaJobConfig(
-      queue: String,
-      jobSetId: String,
+  /** Resolves a value based on precedence: CLI > Template > Default
+    *
+    * @param cliValue
+    *   Value from CLI configuration
+    * @param templateValue
+    *   Value from template
+    * @param defaultValue
+    *   Default value to use if neither CLI nor template provide a value
+    * @return
```

Comment:
Do we need this much documention for a simple one line function?

Reply:
removed excess docs